src/components/AuthenticationProvider.tsx

```diff
@@ -24,6 +25,7 @@ const AuthenticationContext = createContext<AuthenticationContextValueType>(
 
 export const AuthenticationProvider: ChildrenFC = ({ children }) => {
   const [userName, setUserName] = useState<string>();
+  const [userEmail, setUserEmail] = useState<string>();
```

Comment:
Since you don't check anywhere for undefined, you can set the default to an empty string and get rid of the explicit type, saving characters! :D